acados/sim/sim_irk_integrator.h

```diff
@@ -40,48 +40,79 @@ typedef struct
 typedef struct
 {
     /* external functions */
```

Comment:
Good job! Do you check in the sim tests that all possible combinations of options (compute_hessian, compute_forward, etc) work properly?

Reply:
Nope.
As I stated initially, ERK has some strange behaviour. I just checked again `sim_test_hessian`, trying to also turn forward and/or adjoint sensitivities off. And found:
1. `sens_hess = true, sens_adj = true, sens_forw = false`: produces adjoints that are off
2. `sens_hess = true, sens_adj = false, sens_forw =true `: produces Segmentation fault
3. `sens_hess = true, sens_adj = false, sens_forw = false `: produces Segmentation fault
4. `sens_hess = true, sens_adj = false, sens_forw = false `:  works but hessian and adjoint result are a bit off..

Only 4. is active in the test :arrow_right: test passes
I could instead remove ERK from the test and test all the combinations for IRK..

Maybe I should write an issue for that..

@roversch I would be really curious how your scheme for the paper works if you use the IRK hessians instead of the one from ERK!

Reply:
So IRK passes all 4? If yes I would say: run 1-4 for IRK, only 4 for ERK and more importantly: open an issue to fix 1-3 for ERK.